stock_location_restrict_group/models/stock_location.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockLocation(models.Model):
+
+    _inherit = 'stock.location'
+
+    group_restricted = fields.Boolean(
+        'Group Restricted',
+        help="Check this if you want to restrict transfers to one procurement"
+        " group only")
+    restricted_group = fields.Many2one(
+        'procurement.group',
+        compute='_compute_restricted_group')
+    restricted_group_name = fields.Char(
+        related='restricted_group.name')
+
+    @api.multi
+    def _compute_restricted_group(self):
+        # Looking for quants located in locations of 'group' type
+        # And then look into linked moves or pack operations
+        for location in self.filtered(lambda l: l.group_restricted):
+            quants = self.env['stock.quant'].search(
+                [('location_id', '=', location.id)])
+            location.restricted_group = False
```

Comment:
If you already have a restricted_group for the location, why to delete it and recompute it?